awspub/image_marketplace.py

```diff
@@ -43,11 +44,14 @@ def request_new_version(self, image_id: str) -> None:
 
         # version doesn't exist already - create a new one
         changeset = self._request_new_version_changeset(image_id)
+        # changeset names cannot include some characters valid for a version_title. Remove all invalid characters
```

Comment:
With regex I tend to think it's a good idea to leave a comment describing as precisely as possible the intent of the expression. In the spirit of that principle, it could be a good idea to state in the comment exactly which characters are invalid, in plain language, so that a reviewer or maintainer can ensure the regex is doing what it's meant to do as support stories (such as AWS handling of characters) change.

Reply:
I don't have a concrete list of disallowed characters, but the regex is just the negated list of all allowed characters. So `\w\s+=.:@-` represents every valid character for the changeset name. I can update the comment to make that more clear.

Reply:
adding https://docs.aws.amazon.com/marketplace-catalog/latest/api-reference/API_StartChangeSet.html#API_StartChangeSet_RequestSyntax as reference to the comment would be good

Reply:
updated the comment and added a link to the docs